### PR TITLE
Export api

### LINKED ISSUE #157: Add ability to sort by field within data management table
**What is the expected state?**
I expect to be able to click on any individual column header on the data management table and have the table be sorted by that column. If I click the same column again, the sort should invert. If I click another column header the table should instead sort by that column value.

**What is the actual state?**
If I click on any column within the data management table, no sorting occurs.

**Relevant context**
`va_explorer/va_data_management/views.py`

### LINKED ISSUE #167: Support programmatic data export
**Is your feature request related to a problem? Please describe.**
As an admin who works with other systems, I am limited to the download button in terms of exporting data from VA Explorer and can't easily share VA Explorer data with other systems such as the DNRPC's in Zambia

**Describe the solution you'd like**
I would like VA Explorer to provide some endpoint which exports its data in a given or documented format that other systems accept or can query. Ideally this export can be automated for systems that frequently share data.

**Describe alternatives you've considered**
- Determine other systems desired format and input endpoint then implement a function to send the data to that endpoint (assumes only 1 other system, may be infeasible for multiple)
- Provide a passive queryable endpoint along with documentation for systems that would like access

**Additional context**
- Methods of authentication will need to be considered

### PR DESCRIPTION
**Pull request by babraham33**
This request primarily closes #167 by adding an api to download data, a download form view that connects to it, and a suite of tests to validate download functionality

To test: 
Go to Download Data tab, fill out form in a way that filters data, download, and confirm data meets filter criteria

Also closes #157

### CODE REVIEW

File: va_explorer/templates/partials/_top_navigation.html [hunk 1]
```diff
@@ -30,6 +30,11 @@
           <a class="{% active '\/(.*)dashboard' %} nav-link" href="{% url 'va_analytics:dashboard' %}">Dashboard</a>
       </li>
       {% endif %}
+      {% if perms.va_analytics.download_data %}
+      <li class="nav-item">
+          <a class="{% active '\/(.*)download' %} nav-link" href="{% url 'va_export:download_form' %}">Download Data</a>
```
Comment (ajbarnes) [inline va_explorer/templates/partials/_top_navigation.html:35]: Thoughts on calling this "Export" instead?

File: va_explorer/templates/va_export/index.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{% load static %}
+{% load crispy_forms_tags %}
+
+{% block title %}VA Export{% endblock %}
+{% block heading %}Customize VA Export{% endblock %}
+
+
+{% block content %}
+<div class = "col-6">
+  <p>Customize which data you'd like to export and how. First, choose whether you want to export to an external databse (still a WIP) or download locally. Then, choose the timeframe and geographic scope of your data. </p>
```
Comment (ajbarnes) [inline va_explorer/templates/va_export/index.html:11]: Thoughts on using "(Coming Soon)" instead of "(still a WIP)"

File: va_explorer/va_analytics/dash_apps/va_dashboard.py [hunk 3]
```diff
@@ -538,11 +540,13 @@ def init_va_data(hidden_trigger=None, **kwargs):
...
     # Only show the Download Data button if user has access to it.
-    download_div = html.Div(id="download_data_button")
-    if kwargs["user"].can_download_data:
-        download_div.children = [
-            dbc.Button("Download Data", href="/va_analytics/download", color="primary", external_link=True),
-        ]
+    download_div = html.Div(id="download_data_div")
+    download_div.children = [
+        dbc.Button("Download Data", id="download_button", href=reverse("va_export:va_api"), color="primary",
+            external_link=True) #title="Download data meeting dashboard filter criteria")
```
Comment (ajbarnes) [inline va_explorer/va_analytics/dash_apps/va_dashboard.py:546]: Is this comment still needed?

File: va_explorer/va_analytics/urls.py [hunk 4]
```diff
@@ -5,13 +5,13 @@
 
 from va_explorer.va_analytics.views import (
     dashboard_view, 
-    download_csv,
+    # download_csv,
```
Comment (ajbarnes) [inline va_explorer/va_analytics/urls.py:8]: Is this no longer needed?

File: va_explorer/va_analytics/views.py [hunk 5]
```diff
@@ -27,66 +27,66 @@ class DashboardView(CustomAuthMixin, PermissionRequiredMixin, TemplateView):
...
-        response["Content-Disposition"] = 'attachment; filename="va_download.csv"'
-        va_df.to_csv(response, index=False)
-
-        write_va_log(LOGGER, f"[download] clicked download data", self.request)
-
-        return response
-
-
-download_csv = DownloadCsv.as_view()
+# class DownloadCsv(CustomAuthMixin, PermissionRequiredMixin, View):
```
Comment (ajbarnes) [inline va_explorer/va_analytics/views.py:30]: Consider removing if this is going away completely.

File: va_explorer/va_export/views.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+from va_explorer.va_logs.logging_utils import write_va_log
+from va_explorer.va_data_management.utils.location_assignment import fuzzy_match
+from va_explorer.va_export.forms import VADownloadForm
+from va_explorer.va_export.utils import get_loc_ids_for_filter
+
+
+
+LOGGER = logging.getLogger("event_logger")
+
+# TODO: REMOVE THIS AND IMPLEMENT AUTH
```
Comment (ajbarnes) [inline va_explorer/va_export/views.py:37]: #178 opened to handle this in the future, for now we can uncomment the regular auth/permissions, right?

File: va_explorer/va_logs/admin.py [hunk 7]
```diff
@@ -1,3 +1,7 @@
 from django.contrib import admin
+from .models import VerbalAutopsy, Location, CauseOfDeath
 
 # Register your models here.
+admin.site.register(VerbalAutopsy)
```
Comment (ajbarnes) [inline va_explorer/va_logs/admin.py:5]: Are these changes needed? If so, could you expand on what they do for this feature?

File: va_explorer/va_logs/models.py [hunk 8]
```diff
@@ -1,3 +1,10 @@
 from django.db import models
 
-# Create your models here.
+# import models already created in va data management
+from va_explorer.va_data_management.models import VerbalAutopsy, Location, CauseOfDeath
```
Comment (ajbarnes) [inline va_explorer/va_logs/models.py:4]: Similar to admin.py, are these changes needed? If so, could you expand on what they do for this feature?

File: va_explorer/va_data_management/tests/test_views.py [hunk 9]
```diff
@@ -302,6 +302,9 @@ def test_field_worker_access_control():
 
     response = client.get("/va_data_management/")
     assert response.status_code == 200
+
+    # import pdb; pdb.set_trace()
```
Comment (ajbarnes) [inline va_explorer/va_data_management/tests/test_views.py:306]: Stray debugger statement, slated for removal

**Review by ajbarnes — COMMENTED** [review]
Reviewed the code and left some comments with my thoughts. Pulled locally and successfully ran the tests after fixing a small bug. I was also able to replicate the described behavior in the UI on both the new export page and on the dashboard. I say LGTM after the comments are addressed.

File: va_explorer/templates/partials/_top_navigation.html (hunk 1 above)
Comment (babraham33) [reply va_explorer/templates/partials/_top_navigation.html]: Changed to Export

**Review by ajbarnes — APPROVED** [review]

File: va_explorer/va_analytics/views.py (hunk 5 above)
Comment (babraham33) [reply va_explorer/va_analytics/views.py]: removed

File: va_explorer/va_analytics/urls.py (hunk 4 above)
Comment (babraham33) [reply va_explorer/va_analytics/urls.py]: no longer needed

File: va_explorer/va_logs/admin.py (hunk 7 above)
Comment (babraham33) [reply va_explorer/va_logs/admin.py:5]: no, removed
